tests/test_api.py

```diff
@@ -162,10 +169,31 @@ def test_format_edtf_format():
         '2020-02-29/2020-12-31',
         Interval(Date('2020', '02', '29'), Date('2020', '12', '31'))
     ),
+    # ## Dates and times
+    # Complete representations for calendar date and (local) time of day
+    (
+        '1985-04-12T23:20:30',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30')
+    ),
+    # Complete representations for calendar date and UTC time of day
+    (
+        '1985-04-12T23:20:30Z',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30Z')
+    ),
+    # Date and time with timeshift in hours (only)
+    (
+        '1985-04-12T23:20:30-04',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30-04')
+    ),
+    # Date and time with timeshift in hours and minutes
+    (
+        '1985-04-12T23:20:30+04:30',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30+04:30')
```

Comment:
Running these tests with version `v4.0.1` of `edtf` was failing with the following, while upgrading to `v5.0.0` fixes the bug.

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
babel_edtf/__init__.py:213: in parse_edtf
    edtf_date = edtf_parse_edtf(date)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

str = '2004-01-01T23:20:30-04:00', parseAll = True, fail_silently = False

    def parse_edtf(str, parseAll=True, fail_silently=False):
        try:
            if not str:
                raise ParseException("You must supply some input text")
            p = edtfParser.parseString(str.strip(), parseAll)
            if p:
                return p[0]
        except ParseException as e:
            if fail_silently:
                return None
>           raise EDTFParseException(e)
E           edtf.parser.edtf_exceptions.EDTFParseException: Expected end of text, found 'T23'  (at char 10), (line:1, col:11)  (at char 0), (line:1, col:1)

.venv/lib/python3.9/site-packages/edtf/parser/grammar.py:291: EDTFParseException

========================================================================================================== short test summary info ===========================================================================================================
FAILED tests/test_api.py::test_parse_edtf[2004-01-01T23:20:30-04:00-expected8] - edtf.parser.edtf_exceptions.EDTFParseException: Expected end of text, found 'T23'  (at char 10), (line:1, col:11)  (at char 0), (line:1, col:1)
================================================================================================== 1 failed, 76 passed, 15 skipped in 1.56s ==================================================================================================
```